heartbeat/tests/system/test_autodiscovery.py

```diff
@@ -0,0 +1,62 @@
+import os
+from heartbeat import BaseTest
+import unittest
+import re
+
+from beat.beat import INTEGRATION_TESTS
+
+
+class TestAutodiscover(BaseTest):
+    """
+    Test heartbeat autodiscover
+    """
+    @unittest.skipIf(not INTEGRATION_TESTS or
+                     os.getenv("TESTING_ENVIRONMENT") == "2x",
+                     "integration test not available on 2.x")
+    def test_docker(self):
+        """
+        Test docker autodiscover starts modules from templates
+        """
+        import docker
+        docker_client = docker.from_env()
+
+        self.render_config_template(
+            autodiscover={
+                'docker': {
+                    'templates': '''
+                      - condition:
+                          contains:
+                            docker.container.image: redis
+                        config:
+                          - type: tcp
+                            hosts: ["${data.host}:${data.port}"]
+                            schedule: "@every 1s"
+                            timeout: 1s
+                    ''',
+                },
+            },
+        )
+
+        proc = self.start_beat()
+
+        self.wait_until(lambda: self.log_contains(re.compile('autodiscover.+Got a start event:', re.I)))
+
+        self.wait_until(lambda: self.output_count(lambda x: x >= 1))
+        container.stop()
```

Comment:
This seems the line it fails. As redis is started outside, no need to stop the container here. I assume varialbe is not initialised?